tempodb/encoding/vparquet/block_search.go

```diff
@@ -317,14 +422,75 @@ func rawToResults(ctx context.Context, pf *parquet.File, rgs []parquet.RowGroup,
 	return results
 }
 
-type parquetSearchMetrics struct {
-	// TODO:  this isn't accurate, figure out a good way to measure this
-	//pTraceID *pq.InstrumentedPredicate
+// searchStandardTagValues searches a parquet file for "standard" tags. i.e. tags that don't have unique
+// columns and are contained in labelMappings
+func searchStandardTagValues(ctx context.Context, tag string, pf *parquet.File, cb common.TagCallback) error {
+	rgs := pf.RowGroups()
+	makeIter := makeIterFunc(ctx, rgs, pf)
+
+	keyPred := pq.NewStringInPredicate([]string{tag})
+
+	iter := pq.NewJoinIterator(DefinitionLevelResourceAttrs, []pq.Iterator{
+		makeIter(FieldResourceAttrKey, keyPred, "keys"),
+		makeIter(FieldResourceAttrVal, nil, "values"),
+	}, nil)
+	for {
+		match := iter.Next()
+		if match == nil {
+			break
+		}
+		m := match.ToMap()
```

Comment:
It's possible to iterate `match.Entries` looking for `.Key=="values"`, which should be more efficient than calling `.ToMap`.   Interestingly, I don't see that done anywhere in the current code base, but it is a common pattern in the upcoming traceql implementation (so I would consider it an acceptable approach).

Reply:
I think I got it from the method above `rawToResults`.